src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -104,10 +104,10 @@ public static void Serealize_NewSlotPublicProperty_SpecifiedJsonPropertyName()
             var obj = new ClassWithNewSlotAttributedDecimalProperty();
             string json = JsonSerializer.Serialize(obj);
 
-            Assert.Equal(@"{""MyNumeric"":1,""MyNewNumeric"":1.5}", json);
+            Assert.Equal(@"{""MyNewNumeric"":1.5,""MyNumeric"":1}", json);
```

Comment:
Not sure we can guarantee the order of the JSON here. Even though we add properties on more derived types to `_properyCache` first, the order isn't deterministic when we copy that over to `_propertyCacheArray` (used on serialization). We should use `Assert.Contains` here and anywhere else the tests assume a specific order (more than two properties serialized).